docs/source/en/generation_strategies.md

```diff
@@ -465,30 +484,35 @@ If the main and assistant models have different tokenizers, use Universal Assist
 ['Alice and Bob are sitting in a bar. Alice is drinking a beer and Bob is drinking a']
 ```
 
-When using assisted decoding with sampling methods, you can use the `temperature` argument to control the randomness,
-just like in multinomial sampling. However, in assisted decoding, reducing the temperature may help improve the latency.
+#### Prompt Lookup
+
+Alternatively, you can also set the `prompt_lookup_num_tokens` to trigger n-gram based assisted decoding, as opposed
+to model based assisted decoding. You can read more about it [here](https://twitter.com/joao_gante/status/1747322413006643259).
+
+#### Self-Speculative Decoding
+
+An LLM can be trained to also use its language modeling head with earlier hidden states as input, effectivelly
+skipping layers to yield a lower-quality output -- a technique called early exits.
+If the model you're using was trained to do early exit, you can pass
+`assistant_early_exit` (integer). In this case, the assistant model will be the model itself with early exit, hence
+"self-speculative" name. Contrarily to vanilla assisted generation, here the assistant model shares the cache with the
+model itself, since they are the same model, resulting in lower memory requirements.
```

Comment:
```suggestion
model itself, since they are the same model, resulting in lower memory requirements.

In the snippet below, the self-speculative assistant is exiting after the first four layers.
```